cylc/flow/network/schema.py

```diff
@@ -412,10 +412,12 @@ async def get_nodes_by_ids(root, info, **args):
             parent_args.update(
                 {'id': parent_id, 'delta_store': False}
             )
-            parent = await resolvers.get_node_by_id(
-                NODE_MAP[get_type_str(info.parent_type)],
-                parent_args
-            )
+            parent_type = get_type_str(info.parent_type)
+            if parent_type in NODE_MAP:
+                parent = await resolvers.get_node_by_id(
+                    NODE_MAP[parent_type], parent_args)
+            else:
+                parent = await resolvers.get_workflow_by_id(parent_args)
```

Comment:
Right, so if we don't have the parent in the map/cache, then we define the `parent` as the workflow (which is retrieved with the `parent_args`)? It won't ever happen that we have a parent missing that is some other type, like a family, or task?

Don't know much about the code involved here, but looks OK to me.

Reply:
(also not sure if it'd be easy to write a test for this, I guess it'd be a bit hard?)

Reply:
> @dwsutherland could you just explain briefly what the effect of this bug was, and how to trigger this bit of code (I'm having trouble doing that, via a debugger, so I can observer the content of the variables here...)

Run this query in GraphiQL on a stopped workflow:
```
{
  workflows (ids: ["nox"]) {
    id
    taskProxies {
      id
    }
  }
}
```
and you'll get:
```
{
  "errors": [
    {
      "message": "'Workflow'",
      "locations": [
        {
          "line": 4,
          "column": 5
        }
      ],
      "path": [
        "workflows",
        0,
        "taskProxies"
      ]
    }
  ],
  "data": {
    "workflows": [
      {
        "id": "sutherlander|nox",
        "taskProxies": null
      }
    ]
  }
}
```

Reply:
Probably would be good as a functional test when the CLI can go through the UIS.